### ISSUE TITLE
Beta: undefined class when using decorate and derive

### ISSUE THREAD

**Issue by nxht** [elysiajs/elysia#503]
### What version of Elysia.JS is running?

1.0.0-beta.13

### What platform is your computer?

Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

- Class is undefined error when using decorated class during derive in a plugin
- Minimal reproduction code
```typescript
import { Elysia } from "elysia";


class AClass {
  run() {
    return 1
  }
}

const p = new Elysia().decorate('a', new AClass()).derive(({a}) => {
  const b = a.run()
  return {b}
})


const app = new Elysia().use(p)
  .get("/", ({b}) => `hi, ${b}`)
  .listen(3000);

console.log(
  `http://${app.server?.hostname}:${app.server?.port}`
);
```

### What is the expected behavior?

```hi 1```

### What do you see instead?

```json
{
  "name": "TypeError",
  "message": "a.run is not a function. (In 'a.run()', 'a.run' is undefined)"
}
```

### Additional information

- Works fine on stable version but errors in both `beta.13` and `exp.14`

**Comment by SaltyAom** [comment]
Fixed in [711925a](https://github.com/elysiajs/elysia/pull/465/commits/711925a5b5561452b58979b26917cb13a6cf8d9c) under `1.0.0-beta.14`.

Please let me know if the issue still happening.

**Comment by nxht** [comment]
@SaltyAom 
Works fine now. Thanks for the fix
